redfish-core/lib/account_service.hpp

```diff
@@ -130,6 +132,190 @@ inline std::string getPrivilegeFromRoleId(std::string_view role)
     return "";
 }
 
+inline bool getAccountTypeFromUserGroup(std::string_view userGroup,
+                                        nlohmann::json& accountTypes)
+{
+    // set false if userGroup values are not found in list, return error
+    bool isFoundUserGroup = true;
+
+    if (userGroup == "redfish")
+    {
+        accountTypes.push_back("Redfish");
+    }
+    else if (userGroup == "ipmi")
+    {
+        accountTypes.push_back("IPMI");
+    }
+    else if (userGroup == "ssh")
+    {
+        accountTypes.push_back("HostConsole");
+        accountTypes.push_back("ManagerConsole");
+    }
+    else if (userGroup == "web")
+    {
+        accountTypes.push_back("WebUI");
+    }
+    else
+    {
+        // set false if userGroup not found
+        isFoundUserGroup = false;
+    }
+
+    return isFoundUserGroup;
+}
+
+inline bool getUserGroupFromAccountType(
+    const std::optional<std::vector<std::string>>& accountTypes,
+    std::vector<std::string>& userGroup)
+{
+    // set false if AccountTypes values are not found in list, return error
+    bool isFoundAccountTypes = true;
+
+    bool isRedfish = false;
+    bool isIPMI = false;
+    bool isHostConsole = false;
+    bool isManagerConsole = false;
+    bool isWebUI = false;
+
+    for (const auto& accountType : *accountTypes)
+    {
+        if (accountType == "Redfish")
+        {
+            isRedfish = true;
+        }
+        else if (accountType == "IPMI")
+        {
+            isIPMI = true;
+        }
+        else if (accountType == "WebUI")
+        {
+            isWebUI = true;
+        }
+        else if ((accountType == "HostConsole"))
+        {
+            isHostConsole = true;
+        }
+        else if (accountType == "ManagerConsole")
+        {
+            isManagerConsole = true;
+        }
+        else
+        {
+            // set false if accountTypes not found and return
+            isFoundAccountTypes = false;
+            return isFoundAccountTypes;
+        }
+    }
+
+    if (isRedfish)
+    {
+        userGroup.emplace_back("redfish");
+    }
+    if (isIPMI)
+    {
+        userGroup.emplace_back("ipmi");
+    }
+    if (isWebUI)
+    {
+        userGroup.emplace_back("web");
+    }
+    if ((isHostConsole) ^ (isManagerConsole))
+    {
+        BMCWEB_LOG_ERROR << "HostConsole or ManagerConsole, one of value is "
+                            "missing to set SSH property";
```

Comment:
You could send Redfish message StrictAccountTypes for this condition.
https://github.com/openbmc/bmcweb/blob/master/redfish-core/include/registries/base_message_registry.hpp
Background: https://github.com/openbmc/bmcweb/blob/master/static/redfish/v1/schema/ManagerAccount_v1.xml

Reply:
https://redfish.dmtf.org/registries/Base.1.12.1.json
```
       "StrictAccountTypes": {
            "Description": "Indicates the request failed because a set of `AccountTypes` or `OEMAccountTypes` was not accepted while `StrictAccountTypes` is set to `true`.",
            "LongDescription": "This message shall be used to indicate that the request failed because a set of `AccountTypes` or `OEMAccountTypes` was not accepted while `StrictAccountTypes` is set to `true`.",
            "Message": "The request was not possible to fulfill with the account types included in property '%1' and property StrictAccountTypes set to true.",
            "Severity": "Warning",
            "MessageSeverity": "Warning",
            "VersionAdded": "1.9.0",
            "NumberOfArgs": 1,
            "ParamTypes": [
                "string"
            ],
            "ArgDescriptions": [
                "The name of the property."
            ],
            "ArgLongDescriptions": [
                "This argument shall contain the name of the property `AccountTypes` or `OEMAccountTypes`."
            ],
            "Resolution": "Resubmit the request either with an acceptable set of AccountTypes and OEMAccountTypes or with StrictAccountTypes set to false."
        },
```

We should also put out that StrictAccountTypes and just hardcode to True since we are enforcing it

Reply:
Agreed.  Change GET /redfish/v1/AccountService property {"StrictAccountTypes": true}